devcycle_python_sdk/managers/event_queue_manager.py

```diff
@@ -1,28 +1,111 @@
-import json
+import threading
+import time
 import logging
+import json
 
 from devcycle_python_sdk.dvc_options import DevCycleLocalOptions
 from devcycle_python_sdk.api.local_bucketing import LocalBucketing
+from devcycle_python_sdk.api.event_client import EventAPIClient
+from devcycle_python_sdk.exceptions import (
+    APIClientError,
+    APIClientUnauthorizedError,
+    NotFoundError,
+)
+from devcycle_python_sdk.models.event import FlushPayload, EventType
+
 
 logger = logging.getLogger(__name__)
 
 
-class EventQueueManager:
+class EventQueueManager(threading.Thread):
     def __init__(
         self,
         sdk_key: str,
         options: DevCycleLocalOptions,
         local_bucketing: LocalBucketing,
     ):
-        self.sdk_key = sdk_key
-        self.options = options
-        self.local_bucketing = local_bucketing
+        super().__init__()
+
+        if sdk_key is None or sdk_key == "":
+            raise ValueError("DevCycle is not yet initialized to publish events.")
+
+        self._sdk_key = sdk_key
+        self._options = options
+        self._local_bucketing = local_bucketing
+        self._event_api_client = EventAPIClient(self._sdk_key, self._options)
+        self._flush_lock = threading.Lock()
+
+        # Setup the event queue inside the WASM module
+        event_options_json = json.dumps(self._options.event_queue_options())
+        self._local_bucketing.init_event_queue(event_options_json)
+
+        # Only start event processing thread if event logging is enabled
+        if not (
+            self._options.disable_custom_event_logging
+            and self._options.disable_automatic_event_logging
+        ):
+            self._processing_enabled = True
+            self.daemon = True
+            self.start()
+        else:
+            self._processing_enabled = False
+
+    def _flush_events(self) -> int:
+        if self._flush_lock.locked():
+            return 0
+
+        with self._flush_lock:
+            try:
+                payloads = self._local_bucketing.flush_event_queue()
+            except Exception as e:
+                logger.error(f"Error flushing event payloads: {str(e)}")
+
+            event_count = 0
+            if payloads:
+                logger.info(f"DVC Flush {len(payloads)} event payloads")
+                for payload in payloads:
+                    event_count += payload.eventCount
+                    self._publish_event_payload(payload)
+                logger.info(
+                    f"DVC Flush {event_count} events, for {len(payloads)} users"
+                )
+            return event_count
+
+    def _publish_event_payload(self, payload: FlushPayload) -> None:
+        if payload and payload.records:
+            try:
+                self._event_api_client.publish_events(payload.records)
+                self._local_bucketing.on_event_payload_success(payload.payloadId)
+            except (APIClientUnauthorizedError, NotFoundError):
```

Comment:
Should we have a different message for NotFoundError? Seems like that would actually indicate that the event URL was wrong.